.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow file builds and pushes the Dockerfile
+# located in this repo, to the SCD harbor Docker registry.
+# It uses a harbor robot account, who's credentials are 
+# stored as secrets in this repo.
+
+name: Build & Push Docker Image
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - deployment/push-docker-image-to-harbor-#1030
```

Comment:
Looks good. I think pushing branch `push-docker-image-to-harbor-#1030`  is fine for testing purposes, but at some stage we should standardize all the action files to only push the `k8s-deployment` branch, or maybe anything under deployment, so: `/deployment/*`. This would mean we'd need to rename the [plugin branch](https://github.com/ral-facilities/datagateway/tree/feature/k8s-deployment) though, as it's still using the `feature` tag.